components/eventing-controller/api/v1alpha2/subscription_webhook.go

```diff
@@ -115,11 +117,24 @@ func (s *Subscription) validateSubscriptionTypes() *field.Error {
 	return nil
 }
 
-func (s *Subscription) validateSubscriptionConfig() *field.Error {
+func (s *Subscription) validateSubscriptionConfig() field.ErrorList {
+	var allErrs field.ErrorList
 	if isNotInt(s.Spec.Config[MaxInFlightMessages]) {
-		return MakeInvalidFieldError(ConfigPath, s.Name, StringIntErrDetail)
+		allErrs = append(allErrs, MakeInvalidFieldError(ConfigPath, s.Name, StringIntErrDetail))
 	}
-	return nil
+	if s.ifKeyExistsInConfig(ProtocolSettingsQos) && types.IsInvalidQoS(s.Spec.Config[ProtocolSettingsQos]) {
+		allErrs = append(allErrs, MakeInvalidFieldError(ConfigPath, s.Name, InvalidQosErrDetail))
+	}
+	if s.ifKeyExistsInConfig(WebhookAuthType) && types.IsInvalidAuthType(s.Spec.Config[WebhookAuthType]) {
+		allErrs = append(allErrs, MakeInvalidFieldError(ConfigPath, s.Name, InvalidAuthTypeErrDetail))
+	}
+	if s.ifKeyExistsInConfig(WebhookAuthGrantType) && types.IsInvalidGrantType(s.Spec.Config[WebhookAuthGrantType]) {
+		allErrs = append(allErrs, MakeInvalidFieldError(ConfigPath, s.Name, InvalidGrantTypeErrDetail))
+	}
+	if len(allErrs) == 0 {
+		return nil
+	}
+	return allErrs
```

Comment:
I think we can remove the if-statement so just always return `return allErrs`.

Reply:
I think that's right! An empty slice would be nil (need to double check it)

Reply:
done